docs/form-app-script.js

```diff
@@ -0,0 +1,120 @@
+  var formResponses = FormApp.getActiveForm().getResponses();
+  var mostRecentResponse = formResponses[formResponses.length - 1];
+  mostRecentResponse.getItemResponses().forEach(function(item) {
+    attributes[item.getItem().getTitle()] = item.getResponse();
+  });
+
+  attributes["sponsor_id"] = getOrCreateSponsor(attributes).id;
+  attributes["show_id"] = getOrCreatePodcast(attributes).id;
+
+  createCampaign(attributes);
+  setChoices(); // reset the answer choices for exising sponsor / podcast by pulling from jingle
+}
+
+function getOrCreateSponsor(attributes) {
+  if (!attributes['Existing sponsor'].match(/\Other/)) {
+    return {id: attributes['Existing sponsor'].split(':')[1].trim()}
+  } else {
+    var newSponsor = {
+      sponsor: {
+        name: attributes['New sponsor name'],
+        billing_info: attributes['New sponsor billing information'],
+        notes: attributes['Notes about new sponsor']
+      }
+    }
+    var sponsorResponse = JSON.parse(postRequest(SPONSOR_URL, newSponsor).getContentText());
+    return {id: sponsorResponse.id};
+  }
+}
+
+function getOrCreatePodcast(attributes) {
+  if (!attributes['Existing podcast'].match(/\Other/)) {
+    return {id: attributes['Existing podcast'].split(':')[1].trim()}
+  } else {
+    var newPodcast = {
+      show: {
+        name: attributes['New podcast name'],
+        network: attributes['New podcast network'],
+        structure: attributes['New podcast show structure'],
+        recording_day: attributes['New podcast recording day of week (if known)'],
+        rate: attributes['New podcast rate'],
+        notes: attributes['Notes about new podcast']
+      }
+    }
+    var podcastResponse = JSON.parse(postRequest(PODCAST_URL, newPodcast).getContentText());
+    return {id: podcastResponse.id};
+  }
+}
+
+function createCampaign(attributes) {
+  var newCampaign = {
+    campaign: {
+      start_date: attributes['Start date'],
+      end_date: attributes['End date'],
+      copy: attributes['Copy'],
+      zone: attributes['Zone(s)'].join(),
+      sponsor_id: attributes['sponsor_id'],
+      show_id: attributes['show_id']
+    }
+  }
+  postRequest(CAMPAIGN_URL, newCampaign);
+}
+
+function postRequest(url, payload){
+  var request = {
+    method: 'post',
+    contentType: 'application/json',
+    payload:  JSON.stringify(payload)
+  };
+  return UrlFetchApp.fetch(url, request);
+}
+
+// code below handles the setting of the "existing podcast" and "existing sponsor" choices to match what's in Jingle
+// TODO should change these to pull from Adzerk sites + advertizers (networks too)
+
+
+function setChoices() {
+  var questions = FormApp.getActiveForm().getItems();
+  questions.forEach(function (question) {
+    if (question.getTitle() == "Existing sponsor" && question.getType() == FormApp.ItemType.MULTIPLE_CHOICE) {
+      setAnswerChoices(question.asMultipleChoiceItem(), SPONSOR_URL);
```

Comment:
Was it a bug that there are now 4 `Ryan's duck food emporium` sponsors in staging?  (I thought i only submitted that form once).

Reply:
i submitted it a bunch of times trying to debug something (on submit, this pulls the most recent addition to the spreadsheet of form responses and POSTs that, without checking if it exists yet..)

Reply:
Roger.